agents/agent.py

```diff
@@ -265,6 +304,33 @@ def _parse_config_file(self, filename=None):
             if whitelist_commands:
                 self.whitelist_commands.extend(whitelist_commands)
 
+            # APPEND per-instance critical_processes
+            self.logger.debug(
+                'Looking up [server]/critical_processes in %r', filename
+            )
+            try:
+                raw = config.get('server', 'critical_processes')
+                self.logger.debug('Found raw critical_processes: %r', raw)
+                per_instance = parse_csv_list(raw)
+            except (
+                    ConfigParser.NoSectionError, ConfigParser.NoOptionError
+            ) as e:
+                self.logger.warning(
+                    'No per-instance critical_processes in %r: %s', filename, e
+                )
+                per_instance = []
+
+            for proc in per_instance:
+                if proc not in self.critical_processes:
+                    self.logger.debug(
+                        'Appending custom critical_process %r', proc
+                    )
+                    self.critical_processes.append(proc)
+            self.logger.info(
+                'After parsing, critical_processes = %r',
+                self.critical_processes
+            )
```

Comment:
Please see how logging and reading parameters is done in the code above: we use helper functions `maybe_log_message` and `get_config_option`.